### DISCUSSION TITLE
Collision-free planning with second robot in the scene

### DISCUSSION THREAD

**Discussion by shengyangzhuang** [NVlabs/curobo#345] [Research Q&A]
Dear cuRobo team,

I am wondering if cuRobo is able to do collision query between the robot itself and the second additional robot. Ideally, the second robot is also represented with spheres for collision checking. 

If I only used python and ROS2 (I couldn't get cuMotion to work with my kuka iiwa), it seems that I can either use nvblox or import the second robot as a mesh. But I guess this wouldn't have enough accuracy. Apart from that, if the second robot is moving as active constraints, these two ways wouldn't be efficient enough too.

I am appreciated by any comments you may have in this regard. Thank you very much!

With warm regards,

shengyangzhuang

Note: if if can work sth like this?
![image](https://github.com/user-attachments/assets/f5cdcb05-dfbf-496d-ae9f-337ec20ea88c)

**Comment by Sepirence** [comment]
I also try to control multiple robot with curobo. The current solution of curobo that control multiple robots is unifying two robots into one urdf file. It might not be seem to be very exact articulations for both robots, but anyway it works. Look the example, multi_arm_reacher.py. It might be helpful.

**Reply by shengyangzhuang** [reply]
If the second robot is moving, the first robot needs to keep updating its world. How do you send the second robot's mesh/sphere in real time to the first robot for collision query?
